pkg/server/api/agent/v1/service_test.go

```diff
@@ -2,17 +2,217 @@ package agent_test
+			expectedErr: status.Error(codes.InvalidArgument, "ID is not an agent SPIFFE ID"),
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Invalid request: ID is not an agent SPIFFE ID",
+					Data: logrus.Fields{
+						telemetry.SPIFFEID: spiffeid.Must(agentTrustDomain, "agent-1").String(),
+					},
+				},
+			},
+		},
+		{
+			name: "Ban agent fails if agent do not belongs to the server's own trust domain",
+			reqID: &types.SPIFFEID{
+				TrustDomain: "another-example.org",
+				Path:        agentPath,
+			},
+			expectedErr: status.Error(codes.InvalidArgument, "cannot ban an agent that does not belong to this trust domain"),
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Invalid request: cannot ban an agent that does not belong to this trust domain",
+					Data: logrus.Fields{
+						telemetry.SPIFFEID: spiffeid.Must("another-example.org", agentPath).String(),
+					},
+				},
+			},
+		},
+
+		{
+			name: "Ban agent fails if agent does not exists",
+			reqID: &types.SPIFFEID{
+				TrustDomain: agentTrustDomain,
+				Path:        "spire/agent/agent-2",
+			},
+			expectedErr: status.Error(codes.NotFound, "agent not found: rpc error: code = NotFound desc = no such attested node entry"),
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Agent not found",
+					Data: logrus.Fields{
+						logrus.ErrorKey:    "rpc error: code = NotFound desc = no such attested node entry",
+						telemetry.SPIFFEID: spiffeid.Must(agentTrustDomain, "spire/agent/agent-2").String(),
+					},
+				},
+			},
+		},
+		{
+			name: "Ban agent fails if there is a datastore error",
+			reqID: &types.SPIFFEID{
+				TrustDomain: agentTrustDomain,
+				Path:        agentPath,
+			},
+			dsError:     errors.New("unknown datastore error"),
+			expectedErr: status.Error(codes.Internal, "unable to ban agent: unknown datastore error"),
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Unable to ban agent",
+					Data: logrus.Fields{
+						logrus.ErrorKey:    "unknown datastore error",
+						telemetry.SPIFFEID: spiffeid.Must(agentTrustDomain, agentPath).String(),
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test := setupServiceTest(t)
+			defer test.Cleanup()
+			ctx := context.Background()
+
+			node := &common.AttestedNode{
+				SpiffeId:            spiffeid.Must(agentTrustDomain, agentPath).String(),
+				AttestationDataType: "attestation-type",
+				CertNotAfter:        100,
+				NewCertNotAfter:     200,
+				CertSerialNumber:    "1234",
+				NewCertSerialNumber: "1235",
+			}
+
+			_, err := test.ds.CreateAttestedNode(ctx, &datastore.CreateAttestedNodeRequest{
+				Node: node,
+			})
+			require.NoError(t, err)
+
+			test.ds.SetError(tt.dsError)
+			banResp, err := test.client.BanAgent(ctx, &agentpb.BanAgentRequest{Id: tt.reqID})
```

Comment:
can you add a brek line here? just to separate preparation with the call we want to test?